### ISSUE TITLE
Drop indicators didn't show up if "Show window contents while dragging" option is off

### ISSUE THREAD

**Issue by ChaosOptima** [githubuser0xFFFF/Qt-Advanced-Docking-System#179]
Build with Qt5.11.2 in Visual Studio 2017
![hide-window-contents-while-dragging](https://user-images.githubusercontent.com/10622029/82442574-9a2c5000-9aa8-11ea-81cc-f0b93171276c.png)
![docking_issue_2](https://user-images.githubusercontent.com/10622029/82442524-7ff27200-9aa8-11ea-8d32-9344c5a868ff.gif)

**Comment by githubuser0xFFFF** [comment]
Please activate the setting "Show window contents while dragging". If you do not want to show the window contents, please activate non opaque docking.

**Comment by xeash** [comment]
Just wanted to note that tested QtCreator 4.12.1 where they integrated ADS into QML ui design mode, and everything works as expected even with **disabled** "Show window contents while dragging".

**Comment by githubuser0xFFFF** [comment]
Maybe you could investigate a little bit, what the reason is. Maybe non client mouse events are not captured properly?

**Comment by githubuser0xFFFF** [comment]
@xeash Ok, I will try to investigate

**Comment by githubuser0xFFFF** [comment]
@xeash I just tested this with QtCreator 4.12.1 and it also does not work in QtCreator. The ADS reacts on the movement of the floating widget (`void CFloatingDockContainer::moveEvent(QMoveEvent *event)`). If show window content is disabled, then there will be no window movement and no move event.

To sum it up: ADS does not work if **Show window contents while dragging** option is disabled in Windows.

**Comment by xeash** [comment]
@githubuser0xFFFF Yes, you right, I tested incorrectly, and I see same behaviour in Qt Creator. Also same happens with Qt standard QDockWidget.
I was curious and found two apps with dock widgets **where overlay working with Show window contents while dragging disabled**.  Its **Visual Studio** and **Autodesk Maya**. Second one actually use Qt, so I investigated with Spy++ utility(installed with vs studio community) why maya is different.

In Qt Windows plugin source code they handling only WM_MOVE event that maps to QMoveEvent, and we disable "Show window contents while dragging" option then WM_MOVE triggered only when we release the mouse. WM_MOVING event triggered always when we move the window, even with disabled "Show window contents while dragging".  WM_MOVING event is ignored in qt codebase, https://github.com/qt/qtbase/blob/4857fee0fc6a9b277a779266d521d4dc1ca3bd95/src/plugins/platforms/windows/qwindowscontext.cpp#L1117
One workaround I can think of, is install native event filter when we compile on windows platform, and I think this is how VS Studio and Autodesk Maya fix issue.
